test/Interop/Cxx/foreign-reference/Inputs/cxx-functions-and-methods-returning-frt.h

```diff
@@ -0,0 +1,94 @@
+    static FRTStruct* _Nonnull StaticMethodReturningFRT_init() __attribute__((swift_attr("returns_retained")));
+    static FRTStruct* _Nonnull StaticMethodReturningFRT_clone() __attribute__((swift_attr("returns_retained")));
+};
+
+// Struct having static methods returning FRT with attribute swift_attr("returns_unretained") or SWIFT_RETURNS_UNRETAINED
+struct StructWithStaticMethodsReturningFRTWithAttributeReturnsUnretained {
+    static FRTStruct* _Nonnull StaticMethodReturningFRT() __attribute__((swift_attr("returns_unretained")));
+    static FRTStruct* _Nonnull StaticMethodReturningFRT_copy() __attribute__((swift_attr("returns_unretained")));
+    static FRTStruct* _Nonnull StaticMethodReturningFRT_create() __attribute__((swift_attr("returns_unretained")));
+    static FRTStruct* _Nonnull StaticMethodReturningFRT_init() __attribute__((swift_attr("returns_unretained")));
+    static FRTStruct* _Nonnull StaticMethodReturningFRT_clone() __attribute__((swift_attr("returns_unretained")));
+};
+
+
+// Global/free C++ functions returning FRT with both attributes swift_attr("returns_unretained") and swift_attr("returns_retained")
+FRTStruct* _Nonnull global_function_returning_FRT_with_both_attrs_returns_retained_returns_unretained() __attribute__((swift_attr("returns_retained"))) __attribute__((swift_attr("returns_unretained"))); 
+
+// Struct having static method returning FRT with both attributes swift_attr("returns_unretained") and swift_attr("returns_retained")
+struct StructWithStaticMethodsReturningFRTWithBothAttributesReturnsRetainedAndReturnsUnretained {
+    static FRTStruct* _Nonnull StaticMethodReturningFRT() __attribute__((swift_attr("returns_retained"))) __attribute__((swift_attr("returns_unretained")));
+};
+
+struct NonFRTStruct {
+};
+
+// Global/free C++ functions returning non-FRT
+NonFRTStruct* _Nonnull global_function_returning_non_FRT();
+NonFRTStruct* _Nonnull global_function_returning_non_FRT_with_attr_returns_retained() __attribute__((swift_attr("returns_retained")));
+NonFRTStruct* _Nonnull global_function_returning_non_FRT_with_attr_returns_unretained() __attribute__((swift_attr("returns_unretained"))); 
+NonFRTStruct* _Nonnull global_function_returning_non_FRT_create();
+NonFRTStruct* _Nonnull global_function_returning_non_FRT_copy();
+
+
+// Struct having static method returning non-FRT
+struct StructWithStaticMethodsReturningNonFRT {
+    static NonFRTStruct* _Nonnull StaticMethodReturningNonFRT();
+    static NonFRTStruct* _Nonnull StaticMethodReturningNonFRTWithAttrReturnsRetained() __attribute__((swift_attr("returns_retained")));
+    static NonFRTStruct* _Nonnull StaticMethodReturningNonFRTWithAttrReturnsUnretained() __attribute__((swift_attr("returns_unretained")));
+    static NonFRTStruct* _Nonnull StaticMethodReturningNonFRT_create();
+    static NonFRTStruct* _Nonnull StaticMethodReturningNonFRT_copy();
+};
+
```

Comment:
Can you annotate ObjC methods or function pointers with this attribute?

```
@interface C : NSObject
-(FRTStruct*)p __attribute__((swift_attr("returns_retained")));
@end

FRTStruct* (*fp)() __attribute__((swift_attr("returns_retained")));
```

Reply:
Yes! Technically they can be annotated. Whatever in clang languages can be annotated with swift_attr can be annotate with these new returns_retained/returns_unretained attributes. But for objc cases these attributes would be a no-op as per the current implementation. For objc we already have these clang attributes: ns_returns_not_retained/ns_returns_retained .

Reply:
Can `ns_returns_retained` be applied to methods/functions that don't return ObjC objects? 

https://github.com/llvm/llvm-project/blob/main/clang/test/SemaObjC/attr-ns_returns_retained.m

Reply:
@ahatanaka I'll cover the case of ObjC functions and methods returning C++ foreign reference types in a follow-up patch: